build-monitor-acceptance/pom.xml

```diff
@@ -228,6 +229,73 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.codehaus.gmaven</groupId>
+                <artifactId>groovy-maven-plugin</artifactId>
+                <version>2.1.1</version>
+                <executions>
+                    <execution>
+                        <id>download-jenkins-plugins</id>
+                        <phase>pre-integration-test</phase>
+                        <goals>
+                            <goal>execute</goal>
+                        </goals>
+                        <configuration>
+                            <source>
+                                <![CDATA[
+                                    def skipTests = Boolean.valueOf(properties.skipTests)
+                                    def skipITs = Boolean.valueOf(properties.skipITs)
+                                    if (skipTests || skipITs)
+                                        return
+                                    
+                                    def plugins = properties.plugins.split(",")
+                                    def jenkinsVersions = properties.jenkins_versions.split(",") 
+
+                                    def destDir = new java.io.File(properties.dest_dir)
+
+                                    for (def jenkinsVersion : jenkinsVersions) {
+                                        def jenkinsVersionDestDir = new java.io.File(destDir, jenkinsVersion)
+
+                                        for (def pluginId : plugins) {
+                                            def pluginDest = new java.io.File(jenkinsVersionDestDir, pluginId)
+                                            pluginDest.mkdirs()
+                                            def args = [
+                                                "--jenkins-version", jenkinsVersion,
+                                                "--plugins", pluginId,
+                                                "--plugin-download-directory", pluginDest,
+                                                "--war", "${properties.local_repo}/org/jenkins-ci/main/jenkins-war/${jenkinsVersion}/jenkins-war-${jenkinsVersion}.war",
+                                                //"--list",
+                                            ] as String[]
+                                            println "Cache plugin ${pluginId} for jenkins version ${jenkinsVersion}"
+                                            io.jenkins.tools.pluginmanager.cli.Main.main(args)
+                                        }
+                                    }
+                                ]]>
+                            </source>
+                            <properties>
+                                <plugins>build-monitor-plugin,description-setter,workflow-aggregator,badge,cloudbees-folder,external-monitor-job,build-failure-analyzer,claim</plugins>
```

Comment:
I don't think it's a good idea to include `build-monitor-plugin` in this list. That would download the production version of this plugin, which might be dangerous. When using `build-monitor-plugin` from this repository's tests, we should always be using the code under test (i.e. the current build) rather than the production version of this plugin.

Reply:
You're right but it is not actually used in the tests so I just removed it, will also speed up the build so good catch.